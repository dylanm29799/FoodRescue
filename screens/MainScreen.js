import React from "react";
import { View, Text, StyleSheet, Image, FlatList } from "react-native";
import { CATEGORIES } from "../Data/SortDataExample";
import { MaterialIcons } from "@expo/vector-icons";
import { AntDesign } from "@expo/vector-icons";
import Colour from "../constants/Colour";

const MainScreen = (props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.Icons}>
        <MaterialIcons
          style={styles.sort}
          name="sort"
          size={50}
          color="black"
        />
        <Image source={require("../assets/Logo.png")} />
        <AntDesign style={styles.profile} name="user" size={50} color="black" />
      </View>
      <View>
        <Image
          resizeMode={"cover"}
          style={{
            width: "100%",
            height: 300,
            borderWidth: 2,
            borderColor: Colour.secondaryColour,
          }}
          source={require("../assets/Map.png")}
        />
        <Text style={{ fontSize: 40, textAlign: "center" }}>
          {"\n"}Close To You
        </Text>
      </View>

      <View>
        <FlatList></FlatList>
      </View>
    </View>
  );
};
//<Text>{selectedCategory.title}</Text>
//const SortID = props.navigation.getParam("SortID");
//const selectedCategory = CATEGORIES.find((sort) => sort.id === SortID);
const styles = StyleSheet.create({
  screen: {
    flexDirection: "column",
    justifyContent: "space-around",
  },
  Icons: {
    flexDirection: "row",
    alignContent: "center",
  },

  sort: {
    flex: 1,
    justifyContent: "space-around",
  },

  profile: {
    textAlign: "right",
    flex: 1,
    justifyContent: "space-around",
  },
  Map: {},
});

export default MainScreen;
