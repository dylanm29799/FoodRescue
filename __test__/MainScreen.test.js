// __tests__/MainScreen.test.js

import React from "react";
import MainScreen from "../screens/MainScreen";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";
firebase.initializeApp(firebaseConfig);
test("renders correctly", () => {
  const tree = renderer.create(<MainScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
