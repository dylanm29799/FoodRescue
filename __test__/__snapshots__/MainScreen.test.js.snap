// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#c44536",
        "borderWidth": 4.285714285714286,
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <AIRMap
        handlePanDrag={false}
        initialRegion={null}
        onChange={[Function]}
        onLayout={[Function]}
        onMapReady={[Function]}
        onMarkerPress={[Function]}
        region={
          Object {
            "latitude": undefined,
            "latitudeDelta": 0.02,
            "longitude": undefined,
            "longitudeDelta": 0,
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <AIRMapMarker
          coordinate={
            Object {
              "latitude": undefined,
              "longitude": undefined,
            }
          }
          description="Your Location"
          onPress={[Function]}
          stopPropagation={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "position": "absolute",
              },
              undefined,
            ]
          }
          title="You"
        >
          <Text />
        </AIRMapMarker>
      </AIRMap>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "40%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "MonL",
          "fontSize": 42.857142857142854,
          "paddingVertical": 21.428571428571427,
          "textAlign": "center",
        }
      }
    >
      
       
    </Text>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Object {}}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "width": "90%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <Text>
          No Data Found
        </Text>
      </View>
    </RCTScrollView>
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 53.57142857142857,
          "paddingTop": 100,
          "position": "absolute",
          "textAlign": "center",
        }
      }
    >
      Nothing Found, Please try another category
    </Text>
  </View>
</View>
`;
